presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanRepresentation.java

```diff
@@ -73,6 +73,9 @@ public PlanNode getPlanNodeRoot()
 
     public void addNode(NodeRepresentation node)
     {
-        nodeInfo.put(node.getId(), node);
+        NodeRepresentation previous = nodeInfo.put(node.getId(), node);
+        if (previous != null) {
```

Comment:
curious when will we have the duplicated node?

Reply:
It used to happen here:

```
        public Void visitNativeExecution(NativeExecutionNode node, Void context)
        {
            NodeRepresentation nodeOutput = addNode(node, "NativeExecution", ImmutableList.of(node.getSubPlan()));
```

Because NativeExecutionNode has the same ID as the root node of the subplan:

```
    public NativeExecutionNode(PlanNode subPlan)
    {
        this(subPlan.getSourceLocation(), subPlan.getId(), subPlan.getStatsEquivalentPlanNode(), subPlan);
    }
```